components/button/button-add.js

```diff
@@ -109,19 +93,103 @@ class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(Lit
 	render() {
 		const text = this.text || this.localize('components.button-add.addItem');
 		const id = !this.textVisible ? this._buttonId : undefined;
+		const hoverFocusIcon = this._hasHover || this._hasFocus;
+
+		const content = this.textVisible
+			? html`<d2l-button-add-icon-text ?hover-focus-icon="${hoverFocusIcon}" text="${text}"></d2l-button-add-icon-text>`
+			: this._renderWithTextHidden(text, hoverFocusIcon);
 
 		return html`
-			<button class="d2l-label-text" id="${ifDefined(id)}">
+			<button
+				@blur="${this._onBlur}"
+				@focus="${this._onFocus}"
+				@mouseenter="${this._onMouseEnter}"
+				@mouseleave="${this._onMouseLeave}"
+				class="d2l-label-text d2l-visible-on-ancestor-target"
+				id="${ifDefined(id)}"
+				type="button">
+				<div class="line"></div>
+				${content}
 				<div class="line"></div>
-				<div class="content">
-					<d2l-icon icon="tier1:plus-default"></d2l-icon>
-					${this.textVisible
-		? html`<span>${text}</span>`
-		: html`<d2l-tooltip class="vdiff-target" offset="18" for="${this._buttonId}" for-type="label">${text}</d2l-tooltip>`}
-				</div>
 			</button>
 		`;
 	}
+
+	_onBlur() {
+		this._hasFocus = false;
+	}
+
+	_onFocus() {
+		this._hasFocus = true;
+	}
+
+	_onMouseEnter() {
+		this._hasHover = true;
+	}
+
+	_onMouseLeave() {
+		this._hasHover = false;
+	}
+
+	_renderWithTextHidden(text, hoverFocusIcon) {
+		return html`
+			<d2l-button-add-icon-text ?hover-focus-icon="${hoverFocusIcon}" ?visible-on-ancestor="${this.iconOnlyVisibleOnHoverFocus}"></d2l-button-add-icon-text>
+			<d2l-tooltip class="vdiff-target" offset="18" for="${this._buttonId}" for-type="label">${text}</d2l-tooltip>
+		`;
+	}
+
 }
 customElements.define('d2l-button-add', ButtonAdd);
 
+/**
+ * @ignore
+ */
+class ButtonAddIconText extends VisibleOnAncestorMixin(LitElement) {
+	static get properties() {
+		return {
+			hoverFocusIcon: { type: Boolean, attribute: 'hover-focus-icon' },
+			text: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [visibleOnAncestorStyles, css`
+			:host {
+				--d2l-button-add-icon-text-icon-color: var(--d2l-color-galena);
+				align-items: center;
+				display: flex;
+			}
+			:host([text]) {
+				color: var(--d2l-color-celestine);
+				height: 1.5rem;
+				padding: 0 0.3rem;
+			}
+
+			:host([text]) d2l-icon {
+				color: var(--d2l-color-celestine);
+				padding-inline-end: 0.2rem;
+			}
+			:host(:not([text])) d2l-icon {
+				color: var(--d2l-button-add-icon-text-icon-color);
+				margin: -3px; /** hover/click target */
+				padding: 3px; /** hover/click target */
+			}
+
+			span {
+				font-size: 0.7rem;
+				font-weight: 700;
+				letter-spacing: 0.2px;
+				line-height: 1rem;
+			}`
+		];
+	}
+
+	render() {
+		const icon = this.hoverFocusIcon ? 'tier1:add-filled' : 'tier1:add';
```

Comment:
The other option I played with was having the icon completely set by the main component, but this way seemed cleaner.